### PR TITLE
fix: DB2 spans are tagged as MySQL spans instead of generic DB spans

### PR DESCRIPTION
**Pull request by Angith**
### Bug
DB2 spans are tagged as MySQL spans instead of generic DB spans.

### Cause

In Go Tracer, Connection strings are used to distinguish between SQL-like databases based on the `database/sql` package. 
As MySQL and IBM Db2 connection strings are very similar, the IBM Db2 driver is identified as MySQL DB.

### Fix

The driver name is also considered to differentiate between MySQL and IBM Db2. 

### Drawback

Currently, only the `go_ibm_db` driver is added as the known driver for IBM Db2. If a different driver or driver name is used, it would still identify the DB as MySQL. We can add other IBM Db2 drivers to the known drivers list on request.

### CODE REVIEW

File: instrumentation_sql.go [hunk 1]
```diff
@@ -289,7 +292,14 @@ type DbConnDetails struct {
 	Error        error
 }
 
-func ParseDBConnDetails(connStr string) DbConnDetails {
+func ParseDBConnDetails(connStr, driverName string) DbConnDetails {
+
+	if isDB2driver(driverName) {
+		if details, ok := parseDB2ConnDetailsKV(connStr); ok {
```
Comment (sanojsubran) [inline instrumentation_sql.go:298]: Please handle the parsing failure case also

File: instrumentation_sql.go [hunk 2]
```diff
@@ -430,6 +453,49 @@ func parseMySQLConnDetailsKV(connStr string) (DbConnDetails, bool) {
...
+		case strings.HasPrefix(fieldNorm, "server="):
+			address := field[len("server="):]
+			for i, v := range strings.Split(address, ":") {
+				switch i {
+				case 0:
+					details.Host = v
+				case 1:
+					details.Port = v
+				}
+			}
```
Comment (nithinputhenveettil) [inline instrumentation_sql.go:477]:
```suggestion
			addr := strings.Split(address, ":")
			details.Host, details.Port = addr[0], addr[1]
```

**Review by nithinputhenveettil — CHANGES_REQUESTED** [review]
Minor suggestions only.

File: instrumentation_sql.go (hunk 2 above)
Comment (Angith) [reply instrumentation_sql.go:477]: As port is optional here, the port might not be present in the slice returned by the slice.

File: instrumentation_sql.go (hunk 1 above)
Comment (Angith) [reply instrumentation_sql.go]: Good catch! Done

File: instrumentation_sql.go (hunk 2 above)
Comment (Angith) [reply instrumentation_sql.go]: This code is cleaned up! Please check 😊

File: instrumentation_sql.go [hunk 3]
```diff
@@ -92,11 +95,26 @@ func (drv *wrappedSQLDriver) Open(name string) (driver.Conn, error) {
...
+	if isDB2driver(driverName) {
+		if details, ok := parseDB2ConnDetailsKV(connStr); ok {
+			return details
+		}
+
+		return DbConnDetails{RawString: connStr}
+	}
+
+	return ParseDBConnDetails(connStr)
+
```
Comment (sanojsubran) [inline instrumentation_sql.go:115]: Nitpick: redundant line break in my opinion
```suggestion
```

File: instrumentation_sql.go [hunk 4]
```diff
@@ -306,6 +324,19 @@ func ParseDBConnDetails(connStr string) DbConnDetails {
 	return DbConnDetails{RawString: connStr}
 }
 
+// isDB2driver checks the driver belongs to IBM Db2 database.
```
Comment (sanojsubran) [inline instrumentation_sql.go:327]:
```suggestion
// isDB2driver checks whether the driver belongs to IBM Db2 database.
```

File: instrumentation_sql_test.go [hunk 5]
```diff
@@ -231,9 +232,183 @@ func TestOpenSQLDB(t *testing.T) {
...
+			Host:  "localhost:50000",
+			Port:  "",
+			DB:    "sample",
+			User:  "db2inst1",
+			Stmt:  "TEST QUERY",
+			Error: "",
+		}, data.Tags)
+	})
+}
+
```
Comment (sanojsubran) [inline instrumentation_sql_test.go:407]: Suggestion: If the above tests have differences only in their data part, you can consider moving them to a table driven tests.

**Review by sanojsubran — APPROVED** [review]
Good job @Angith. Thanks for taking this up.

**Review by nithinputhenveettil — APPROVED** [review]

File: instrumentation_sql_test.go (hunk 5 above)
Comment (Angith) [reply instrumentation_sql_test.go:377]: Good Suggestion! Done 👍

**Pull request merged**